Throw exceptions on fixture-file problems, fixes #5213

tkn98 opened the pull request:
<3 Thanks for taking the time to contribute! You're awesome! <3

If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/CONTRIBUTING.md 

**Description (for Contributor and Core Developer)**

If a file giving as fixture parameter is not found within the file-system
or could not be identified as actual fixture then treat it as an exception.

Additionally give an exception message that informs about the cause of it
and which path is the culprit.

Refs:

- Command: pim:installer:load-fixtures

- #5213

- #1878

**Definition Of Done (for Core Developer only)**

| Q | A
| --------------------------------- | ---
| Added Specs                       |
| Added Behats                      |
| Changelog updated                 |
| Review and 2 GTM                  |
| Micro Demo to the PO (Story only) |
| Migration script                  |
| Tech Doc                          |

juliensnz commented on src/Pim/Bundle/InstallerBundle/FixtureLoader/ConfigurationRegistry.php:
```diff
@@ -130,7 +128,12 @@ public function getReader($name, $extension)
     protected function setFixtures(array &$ordered, $filePath)
     {
+        if (!is_file($filePath)) {
+            throw new \InvalidArgumentException(sprintf("Error, not a file: '%s'", $filePath));
```
Could you please use `'Error, not a file: "%s"'` ?

juliensnz commented on src/Pim/Bundle/InstallerBundle/FixtureLoader/ConfigurationRegistry.php:
```diff
@@ -150,6 +153,10 @@ protected function setFixtures(array &$ordered, $filePath)
+            $fixturesFound++;
+        }
+        if (!$fixturesFound) {
+            throw new \UnexpectedValueException(sprintf("Can not identify file as fixture: '%s'", $filePath));
```
same here

juliensnz commented on src/Pim/Bundle/InstallerBundle/FixtureLoader/ConfigurationRegistry.php:
```diff
@@ -150,6 +153,10 @@ protected function setFixtures(array &$ordered, $filePath)
             ];
+            $fixturesFound++;
+        }
+        if (!$fixturesFound) {
```
0 === $fixtureFound

tkn98 commented on src/Pim/Bundle/InstallerBundle/Tests/Unit/FixtureLoader/ConfigurationRegistryTest.php:
```diff
@@ -195,13 +209,20 @@ public function testGetFixtures()
+    {
+        $this->configurationRegistry->getFixtures(
+            [
+                '/tmp/entity1.format1',
```
This one is a faux path to an iexistent file. So it should be fine on any system w/o further changes.

tkn98 commented:
@juliensnz  and @skeleton I addressed the issues you commented on, please take a look if this pleases so far. In case we don't read a good weekend for you two!

nidup commented:
Hi @tkn98 

First, thank you very much for this contribution :tada: 

FYI, the installer bundle has been widely re-worked in v1.6 to use standard imports, it now stops the process if a file is missing and the command running import returns a proper exit code (not 0 when encountering errors or warnings).

FYI, we only merge critical or major bug fixes as patchs in already released minor versions because this kind of behavior change / tech improvement may have impact on production projects applying the last patch.

Don't hesitate to test the "new" behavior on more recent versions to see if it fits your needs. If it's not working for you, don't hesitate to re-open the related issue https://github.com/akeneo/pim-community-dev/issues/5213 to continue the discussion on how it could be improved / changed.

I'm really sorry to close this PR, especially because the improvement is really good, the good thing is that you identified the same limitation that we improved in v1.6 version :)

Best regards.

tkn98 commented:
Right now we're still bound to 1.5 due to version constraints. Would be great to see if Pimgento becomes PHP 7 ready that would help us.

So I'm sorry to see that we can't give back changes upstream for the version in use (which normally makes most sense for community editions).

But I'm glad to see this has improved in 1.6, I'll push forward to that version, so far we're bound to Akeneo 1.5 b/c of PIMGento-2 version constraints (see https://github.com/Agence-DnD/PIMGento-2/issues/72).
